hedera-node/src/main/java/com/hedera/services/txns/contract/ContractUpdateTransitionLogic.java

```diff
@@ -21,77 +21,122 @@
-			var id = op.getContractID();
-			var target = contracts.get().get(fromContractId(id));
-
-			var result = customizerFactory.customizerFor(target, validator, op);
-			var customizer = result.getLeft();
-			if (customizer.isPresent()) {
-				ledger.customize(asAccount(id), customizer.get());
-				txnCtx.setStatus(SUCCESS);
-			} else {
-				txnCtx.setStatus(result.getRight());
-			}
-		} catch (Exception e) {
-			log.warn("Avoidable exception!", e);
-			txnCtx.setStatus(FAIL_INVALID);
+		/* --- Extract gRPC --- */
+		final var contractUpdateTxn = txnCtx.accessor().getTxn();
+		final var op = contractUpdateTxn.getContractUpdateInstance();
+		final var targetId = Id.fromGrpcContract(op.getContractID());
+
+		/* --- Translate the account --- */
+		final var target = accountStore.loadContract(targetId);
+		final var candidate = op.getAdminKey();
+
+		/* --- Conditions --- */
+		final var isSentinel = candidate.equals(IMMUTABILITY_SENTINEL_KEY);
+
+		/* --- Validate --- */
+		if (op.hasAdminKey()) {
+			validateFalse(isValidAdminKey(candidate, isSentinel), INVALID_ADMIN_KEY);
+		}
+		if (op.hasExpirationTime()) {
+			validateFalse(!validator.isValidExpiry(op.getExpirationTime()), INVALID_EXPIRATION_TIME);
+		}
+
+		/* --- Extract --- */
+		final var newAdminKey = getKey(candidate, isSentinel, target);
+		final var newProxy = op.hasProxyAccountID() ? op.getProxyAccountID() : null;
+		final var newAutoRenewPeriod = op.hasAutoRenewPeriod() ? op.getAutoRenewPeriod() : null;
+		final var newExpirationTime = op.hasExpirationTime() ? op.getExpirationTime() : null;
+		final var newMemo = affectsMemo(op) ? getMemo(op) : null;
+
+		/* --- Do the business logic --- */
+		target.updateFromGrpcContract(
+				Optional.ofNullable(newAdminKey),
+				Optional.ofNullable(newProxy),
+				Optional.ofNullable(newAutoRenewPeriod),
+				Optional.ofNullable(newExpirationTime),
+				Optional.ofNullable(newMemo),
+				affectsExpiryOnly(op));
+
+		/* --- Persist the changes --- */
+		accountStore.persistUpdatedContract(target);
+	}
+
+	private boolean isValidAdminKey(final Key candidate, final boolean isSentinelKey) {
+		if (isSentinelKey) {
+			return false;
+		} else {
+			var key = MiscUtils.asUsableFcKey(candidate);
+			return key.isEmpty() || key.get() instanceof JContractIDKey;
 		}
 	}
 
+	private JKey getKey(final Key candidate, final boolean isSentinelKey, final Account target) {
+		return isSentinelKey
+				? new JContractIDKey(target.getId().getShard(), target.getId().getRealm(), target.getId().getNum())
+				: MiscUtils.asUsableFcKey(candidate).orElse(null);
+	}
+
+	//TODO: The op.getMemo() is deprecated. What method should be used when memo is not wrapped?
+	boolean affectsMemo(ContractUpdateTransactionBody op) {
+		return op.hasMemoWrapper() || op.getMemo().length() > 0;
+	}
+
+	private String getMemo(final ContractUpdateTransactionBody op) {
+		return op.hasMemoWrapper() ? op.getMemoWrapper().getValue() : op.getMemo();
+	}
```

Comment:
Do we really need those separate methods?

Reply:
I think, that those two private methods improve the readability of the code.
On the other hand - this was the original approach in the UpdateCustomizerFactory, as implemented by Mr. Tinker:

![image](https://user-images.githubusercontent.com/77110630/137259236-b0cd9029-3de3-4c0c-b2da-f87efa1da6eb.png)

![image](https://user-images.githubusercontent.com/77110630/137259285-0371b3ca-7ae8-4377-b325-087f1e2e4fa5.png)

Reply:
I do think that they are unnecessary